packages/flutter_tools/test/general.shard/unified_analytics_test.dart

```diff
@@ -0,0 +1,128 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:file/memory.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/reporting/unified_analytics.dart';
+import 'package:unified_analytics/src/enums.dart';
+import 'package:unified_analytics/unified_analytics.dart';
+
+import '../src/common.dart';
+import '../src/fakes.dart';
+
+void main() {
+  const String userBranch = 'abc123';
+  const String homeDirectoryName = 'home';
+  const DashTool tool = DashTool.flutterTool;
+
+  late FileSystem fs;
+  late Directory home;
+  late FakeAnalytics analyticsOverride;
+
+  setUp(() {
+    fs = MemoryFileSystem.test();
+    home = fs.directory(homeDirectoryName);
+
+    // Prepare the tests by "onboarding" the tool into the package
+    // by invoking the [clientShowedMessage] method for the provided
+    // [tool]
+    final FakeAnalytics initialAnalytics = FakeAnalytics(
+      tool: tool,
+      homeDirectory: home,
+      dartVersion: '3.0.0',
+      platform: DevicePlatform.macos,
+      fs: fs,
+      surveyHandler: SurveyHandler(
+        homeDirectory: home,
+        fs: fs,
+      ),
+    );
+    initialAnalytics.clientShowedMessage();
+
+    analyticsOverride = FakeAnalytics(
+      tool: tool,
+      homeDirectory: home,
+      dartVersion: '3.0.0',
+      platform: DevicePlatform.macos,
+      fs: fs,
+      surveyHandler: SurveyHandler(
+        homeDirectory: home,
+        fs: fs,
+      ),
+    );
+  });
+
+  group('Unit testing getAnalytics', () {
+    testWithoutContext('Successfully creates the instance for standard branch',
+        () {
+      final Analytics analytics = getAnalytics(
+        runningOnBot: false,
+        flutterVersion: FakeFlutterVersion(),
+        environment: <String, String>{},
+        analyticsOverride: analyticsOverride,
+      );
+
+      expect(analytics.clientId, isNot(NoOpAnalytics.staticClientId),
+          reason: 'The CLIENT ID should be a randomly generated id');
+      expect(analytics is! NoOpAnalytics, true);
```

Comment:
When possible, prefer a specific matcher over expecting against a literal value, as the matchers will give more friendly failure messages (in this example, it will say what type your actual value is)

```suggestion
      expect(analytics, isNot(isA<NoOpAnalytics>()));
```

And throughout this library